src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -228,7 +228,7 @@ internal virtual TypeEnumerator GetTypeEnumerator(Type type, string assemblyFile
                 {
                     if (discoveryOption == TestDataSourceDiscoveryOption.DuringDiscovery)
                     {
-                        if (DynamicDataAttached(sourceLevelParameters, test, tests))
+                        if (DynamicDataAttached(sourceLevelParameters, test, tests, nonRunnableTests))
```

Comment:
Because of the line above, if users ask for discovery of data soruces at execution, we won't see the extra entries although this is not linked to data source. The call should be extracted.

Reply:
updated,